src/Microsoft.ML.Core/Data/ColumnTypeExtensions.cs

```diff
@@ -19,9 +19,19 @@ internal static class ColumnTypeExtensions
             (columnType is TimeSpanType) || (columnType is DateTimeType) || (columnType is DateTimeOffsetType);
 
         /// <summary>
-        /// Zero return means either it's not a key type or the cardinality is unknown.
+        /// Zero return means it's not a key type.
         /// </summary>
-        public static int GetKeyCount(this ColumnType columnType) => (columnType as KeyType)?.Count ?? 0;
+        public static ulong GetKeyCount(this ColumnType columnType) => (columnType as KeyType)?.Count ?? 0;
+
+        /// <summary>
+        /// Sometimes it is necessary to cast the Count to an int. This performs overflow check.
+        /// </summary>
+        public static int ChecktRangeReturnCount(this ColumnType columnType, IExceptionContext ectx = null)
```

Comment:
Wait - why do we have this here, and one for `KeyType`?

Also, type-o `ChecktRangeReturnCount` there seems to be an errant `t` after `Check`.